runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/environment/ExternalEnvironmentFactory.java

```diff
@@ -113,6 +113,7 @@ public RemoteEnvironment createEnvironment(Environment environment, String worke
         ManagedChannelFactory.createDefault().forDescriptor(externalPayload.getEndpoint());
     BeamFnApi.StartWorkerResponse startWorkerResponse =
         BeamFnExternalWorkerPoolGrpc.newBlockingStub(managedChannel)
+            .withWaitForReady()
```

Comment:
What does `withWaitForReady` change, and why do we need it?

Also, I don't think this change belongs in the same PR as the job server changes. Where possible, PRs should do exactly one thing.

Reply:
Good point, it should be separated out in a different PR at least.

My thought on this is similar as the job server request that when runner connects to external service, it is possible that external service may not be up and running yet if we bring up external service together with the main job. In addition, there may be intermittent errors, therefore, `withWaitForReady` seems to be a good candidate here.

I also think that we could provide customizable deadline/timeout here too which requires changes to external_payload model.

What are your thoughts on this?

Reply:
If I understand correctly, adding `withWaitForReady` will make requests that formerly failed fast instead block forever. Therefore, I don't think we should add `withWaitForReady` without also adding a timeout, since otherwise the user gets no feedback about why their pipeline is stalled. And I think the timeout should be off by default, since most deployments don't have a problem with starting the external service ahead of time.

Reply:
Removed this change in the PR, I will follow up separately on this.